src/apps/Permissions/PermissionsView.js

```diff
@@ -0,0 +1,170 @@
+import React, { useCallback, useMemo } from 'react'
+import PropTypes from 'prop-types'
+import {
+  ButtonIcon,
+  ContextMenu,
+  ContextMenuItem,
+  DataView,
+  IconTrash,
+  textStyle,
+  useLayout,
+  useTheme,
+} from '@aragon/ui'
+import { usePermissions } from '../../contexts/PermissionsContext'
+import LocalIdentityBadge from '../../components/IdentityBadge/LocalIdentityBadge'
+import PermissionsIdentityBadge from './PermissionsIdentityBadge'
+
+function PermissionsView({ permissions, onManageRole, heading, showApps }) {
+  const { layoutName } = useLayout()
+  const willRenderEntryChild = useMemo(
+    () => permissions.some(permission => permission.entities.length > 1),
+    [permissions]
+  )
+
+  const fields = [
+    'Action',
+    'On app',
+    { label: 'Assigned to entity', childStart: true },
+    'Managed by',
+  ]
+
+  if (!showApps) {
+    fields.splice(1, 1)
+  }
+
+  return (
+    <DataView
```

Comment:
When clicking on the thrash icon on sub views, the sub view gets closed, not sure if this is expected:
https://www.loom.com/share/213cb7e7b3184ecc997ddd1ffc01fef9

Reply:
Ah yes, it’s something I need to improve in `DataView` :+1: